Support different JVM targets when adding the @Generated annotation.

vRallev opened the pull request:
Fixes #75

gabrielittner commented on compiler/src/main/java/com/squareup/anvil/compiler/codegen/dagger/KotlinPoet.kt:
```diff
@@ -173,15 +177,41 @@ private fun TypeName.wrapInLazy(): ParameterizedTypeName {
+      ?: return null
+
+  return when {
+    targetVersion > JVM_1_8 -> "javax.annotation.processing.Generated"
+    targetVersion == JVM_1_8 -> "javax.annotation.Generated"
+    else -> null
```
Will this return `null` for Android modules that target 1.8? On Android the annotation is never available. Dagger does not emit it in Android projects at all.

vRallev replied:
I'd assume it wouldn't. Do you know how Dagger makes this distinction by any chance?

gabrielittner replied:
It's checking the classpath https://github.com/google/auto/blob/master/common/src/main/java/com/google/auto/common/GeneratedAnnotations.java

vRallev replied:
I definitely can reproduce what you said.

vRallev replied:
Resolved.

gabrielittner replied:
Great, thank you

vRallev commented:
I changed my mind. The annotation is annoying. I don't wanna resolve it during compilation. It adds little value and the comment is good enough for me. I removed the annotation in favor of a comment.
